src/modules/esl-popup/core/esl-popup.ts

```diff
@@ -331,23 +327,30 @@ export class ESLPopup extends ESLToggleable {
   }
 
   /** Actions to execute on activator scroll event. */
-  @listen({auto: false, event: 'scroll', target: ($popup: ESLPopup) => $popup.scrollTargets})
+  @listen({auto: false, group: 'observer', event: 'scroll', target: ($popup: ESLPopup) => $popup.scrollTargets})
   protected _onActivatorScroll(e: Event): void {
     if (this._updateLoopID) return;
     this._updatePosition();
   }
 
-  @listen({auto: false, event: 'transitionstart', target: document.body})
+  @listen({auto: false, group: 'observer', event: 'transitionstart', target: document.body})
   protected _onTransitionStart(): void {
     this._startUpdateLoop();
   }
 
-  @listen({auto: false, event: 'resize', target: window})
+  @listen({auto: false, group: 'observer', event: 'resize', target: window})
   @decorate(rafDecorator)
   protected _onResize(): void {
     this._updatePosition();
   }
 
+  @listen({auto: false, group: 'observer', event: ($popup: ESLPopup) => $popup.REFRESH_EVENT, target: window})
+  protected _onRefresh({target}: Event): void {
+    if (!isElement(target)) return;
+    const {activator, $container} = this;
+    if ($container === target || this.contains(target) || activator && isRelativeNode(activator, target)) this._updatePosition();
```

Comment:
```suggestion
    if ($container === target || this.contains(target) || isRelativeNode(activator, target)) this._updatePosition();
```
isRelativeNode is null safe

Reply:
but

Reply:
but `public get activator(): HTMLElement | null | undefined`

Here is the working code
```suggestion
    if ($container === target || this.contains(target) || isRelativeNode(activator ?? null, target)) this._updatePosition();
```